client_new/src/types/location.rs

```diff
@@ -97,7 +97,17 @@ where
 }
 
 // Derives the counter [`RecordId`] from the given vault path and the counter value.
-pub fn derive_record_id<P>(vault_path: P, counter: usize) -> RecordId
+pub fn derive_record_id<V, R>(vault_path: V, record_path: R) -> RecordId
+where
+    V: AsRef<Vec<u8>>,
+    R: AsRef<Vec<u8>>,
+{
```

Comment:
I'm seeing this `AsRef<Vec<u8>>` pattern frequently in stronghold, but I don't understand it. Why not take `AsRef<[u8]>` instead, which would work just as well? The `Vec<u8>` approach forces users to construct a vector, even though a slice of bytes would technically be sufficient. `Client::vault_exists` is one such example where passing a `&[u8]` wouldn't require any internal changes and be helpful to the caller, so one doesn't have to allocate a vector.

Reply:
Good idea! Changed trait bound to `AsRef<[u8]>` 1c28135.